### ISSUE TITLE
High CPU usage, but only when using Safari

### ISSUE THREAD

**Issue by waltershewmake** [mantinedev/mantine#1968]
### What package has an issue

@mantine/core

### Describe the bug

Note: I selected @mantine/core because I'm not sure which package is causing the issue yet. 

When loading a page built with Mantine in Safari (on both Mac and iPhone), the page very quickly becomes unusable due to spikes in CPU usage. The page freezes, stutters, interactions are slow, newly scrolled sections of the page don't render, etc. When checking Activity Monitor on a Mac, the site frequently spikes up to 100% CPU, even with no interaction on page load.

![Screen Shot 2022-08-01 at 8 42 08 PM](https://user-images.githubusercontent.com/21002090/182436953-569431a0-f3eb-42ea-a1aa-560b3a3816f8.png)

These results are not only found on websites created with Mantine 5 but even on https://mantine.dev. Luckily that means it's easy to reproduce.

Steps to reproduce:
- Using Safari, open https://mantine.dev/
- Interact with the page

I am opening this issue but I intend to look into this myself as well. I'm going to begin by creating a NextJS app with every component rendered, and slowly–by process of elimination–find the package causing the issue.

### What version of @mantine/hooks page do you have in package.json?

5.0.2

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

I think it's highly likely that some component is forced into a continuous re-render cycle causing it to build up tons of CPU usage, but I have no way of knowing right now.

**Comment by waltershewmake** [comment]
I think I may have tracked the issue down to the NumberInput component.

**Comment by waltershewmake** [comment]
#### Works
```jsx
<div>
    {new Array(100).fill(0).map((_, i) => (
        <TextInput key={i}/>
    ))}
</div>
```
#### Works
```jsx
<div>
    {new Array(100).fill(0).map((_, i) => (
        <Checkbox key={i}/>
    ))}
</div>
```
#### Causes the extreme lag we talked about
```jsx
<div>
    {new Array(100).fill(0).map((_, i) => (
        <NumberInput key={i}/>
    ))}
</div>
```

The code creates an array of 100 inputs. That number is just an example. I also tried 50, 20, 5, 3, and 2. All of which produced the same result.

The number input works just fine with only 1 instance though.

**Comment by rtivital** [comment]
The issue is resolved, thanks for investigation!

**Comment by rtivital** [comment]
Fixed in 5.0.3

**Comment by ndmt** [comment]
The problem still seems to exist, or it's just me ?
